DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -288,50 +316,36 @@ def set_ethernet_traffic(self, device, name, pkt_data, traffic_type):
                     Type="fixed", FixedSize=pkt_data.get("frameSize", "512")
                 )
                 config_element.TransmissionControl.update(Type="continuous")
-                ethernet_stack = config_element.Stack.find(StackTypeId="^ethernet$")
+                eth_stack = config_element.Stack.find(StackTypeId="^ethernet$")
                 if "vlanID" in pkt_data:
                     vlan_stack = config_element.Stack.read(
-                        ethernet_stack.AppendProtocol(vlan_template)
+                        eth_stack.AppendProtocol(vlan_template)
                     )
                     ipv4_stack = config_element.Stack.read(vlan_stack.AppendProtocol(ipv4_template))
                 else:
                     ipv4_stack = config_element.Stack.read(
-                        ethernet_stack.AppendProtocol(ipv4_template)
+                        eth_stack.AppendProtocol(ipv4_template)
                     )
-                # update the Mac address
                 if "dstMac" in pkt_data:
-                    dst_mac = ethernet_stack.Field.find(
-                        FieldTypeId="ethernet.header.destinationAddress"
-                    )
-                    dst_mac.Auto = False
-                    dst_mac.update(
-                        ValueType="singleValue", SingleValue="{}".format(pkt_data["dstMac"])
-                    )
+                    self.__update_field(eth_stack.Field.find(FieldTypeId="ethernet.header.destinationAddress"),
+                                        pkt_data["dstMac"])
                 if "srcMac" in pkt_data:
-                    src_mac = ethernet_stack.Field.find(FieldTypeId="ethernet.header.sourceAddress")
-                    src_mac.Auto = False
-                    src_mac.update(
-                        ValueType="singleValue", SingleValue="{}".format(pkt_data["srcMac"])
-                    )
+                    self.__update_field(eth_stack.Field.find(FieldTypeId="ethernet.header.sourceAddress"),
+                                        pkt_data["srcMac"])
                 if "vlanID" in pkt_data:
-                    vlan_id = vlan_stack.Field.find(FieldTypeId="vlan.header.vlanTag.vlanID")
-                    vlan_id.Auto = False
-                    vlan_id.update(
-                        ValueType="singleValue", SingleValue="{}".format(pkt_data["vlanID"])
-                    )
-                # update the Ip address
+                    self.__update_field(vlan_stack.Field.find(FieldTypeId="vlan.header.vlanTag.vlanID"),
+                                        pkt_data["vlanID"])
+                    if "vlanPriority" in pkt_data:
+                        self.__update_field(vlan_stack.Field.find(FieldTypeId="vlan.header.vlanTag.vlanUserPriority"),
+                                            pkt_data["vlanPriority"])
+                ti.Tracking.find()[0].TrackBy = ["trackingenabled0", "sourceDestValuePair0",
```

Comment:
Changes not in https://github.com/dentproject/testing/pull/128 [PR](https://github.com/PLVision/dent-testing/pull/39/commits/b46a05943cdb9806e4f325ff0df0cb0c47a96c24)